core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -50,7 +55,19 @@ public UnsafeSorterSpillReader(
       File file,
       BlockId blockId) throws IOException {
     assert (file.length() > 0);
-    final BufferedInputStream bs = new BufferedInputStream(new FileInputStream(file));
+    long bufferSizeBytes =
+        SparkEnv.get().conf().getSizeAsBytes("spark.unsafe.sorter.spill.reader.buffer.size",
+                                             DEFAULT_BUFFER_SIZE_BYTES);
+    if (bufferSizeBytes > Integer.MAX_VALUE) {
+      // fall back to a sane default value
+      logger.error("Value of config \"spark.unsafe.sorter.spill.reader.buffer.size\" exceeds " +
```

Comment:
warning? because this isn't a fatal error.
Is it reasonable to fall back? I'd imagine that specifying an incredibly large buffer is just an error. What about 0 or negative values?